skimage/feature/util.py

```diff
@@ -40,10 +40,10 @@ def extract(self, image, keypoints):
 
 def plot_matches(
     ax,
+    image0,
     image1,
-    image2,
+    keypoints0,
     keypoints1,
```

Comment:
Hmm, I don't think it's as easy as just replacing them. We need some way to warn users. 

We could use decorator / wrapper would probably allow us to inspect what get's passed to the function appropriately. But it might still be a bit unorthodox in the long run, because regardless of how we do this eventually `image1` will change it's meaning from "first image" to "second image".

I think a better way, in case of this function, would be to rename to `plot_matched_features` with the new API and deprecate the `plot_matches`.

Reply:
> Hmm, I don't think it's as easy as just replacing them. We need some way to warn users.

For sure! I need to use `@deprecate_parameter` (at least).

> We could use decorator / wrapper would probably allow us to inspect what gets passed to the function appropriately. But it might still be a bit unorthodox in the long run, because regardless of how we do this eventually `image1` will change it's meaning from "first image" to "second image".

Right...

> I think a better way, in case of this function, would be to rename to `plot_matched_features` with the new API and deprecate the `plot_matches`.

Good idea! Thanks.